.github/workflows/docs_deploy.yml

```diff
@@ -0,0 +1,266 @@
+name: Documentation
+on:
+  push:
+    branches:
+      - main
+    tags:
+      # Only match non-prerelease tags.
+      - '[0-9]+.[0-9]+.[0-9]'
```

Comment:
If it was a regex I'd want to end this in `$` to assert there is no `rc` or other post notation. I looked at the docs for this: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet and it's not a regex and this should work. This also precludes more than 10 patch versions,  but I don't think we've ever needed that. If we do, we can revisit the docs publishing behavior at that time.